lib/socketclusterclient/reconnect.rb

```diff
@@ -0,0 +1,74 @@
+#
+# Module Reconnect provides Reconnection Support
+#
+# @author Piyush Wani <piyush.wani@amuratech.com>
+#
+module Reconnect
+
+  #
+  # Initializes Reconnection related entities
+  #
+  #
+  #
+  #
+  def initialize_reconnect
+    @reconnect_interval = 2000
+    @max_reconnect_interval = 30000
+    @reconnect_decay = 1
+    @max_attempts = nil
+    @attempts_made = 0
+  end
+
+  #
+  # Adds handler for Reconnection
+  #
+  # @param [Integer] reconnect_interval A interval for reconnection attempt( in MiliSeconds  )
+  # @param [Integer] max_reconnect_interval The Max Limit for reconnection interval (in MiliSeconds)
+  # @param [Integer] reconnect_decay A
+  # @param [Integer] max_attempts The Maximum number of Reconnection Attempts Allowed
+  #
+  #
+  #
+  def set_reconnection_listener(reconnect_interval, max_reconnect_interval, reconnect_decay, max_attempts)
+    @max_reconnect_interval = max_reconnect_interval
```

Comment:
Please put this line below as it changes the code output. Refer: https://github.com/sacOO7/socketcluster-client-java/blob/master/src/main/java/io/github/sac/ReconnectStrategy.java

Reply:
It is not needed as the logical bug here but for the sake of micro optimisation, I am changing it.